analyzers/src/SonarAnalyzer.Common/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -18,29 +18,27 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System.Collections.Concurrent;
 using System.Runtime.CompilerServices;
 using Roslyn.Utilities;
 
 namespace SonarAnalyzer.Extensions;
 
 internal static class SyntaxTreeExtensions
 {
-    private static readonly ConditionalWeakTable<Compilation, ConcurrentDictionary<SyntaxTree, bool>> GeneratedCodeCache = new();
+    private static readonly ConditionalWeakTable<SyntaxTree, object> GeneratedCodeCache = new();
```

Comment:
Key and Value in a CWT have the `class` constraint:
https://learn.microsoft.com/de-de/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2?view=net-7.0
Therefore, we store a boxed bool.